src/sampler.jl

```diff
@@ -37,6 +37,9 @@ function adapt!(
     θ::AbstractVector{T},
     α::T
 ) where {T<:Real}
+    if i == 1
```

Comment:
The code looks correct to me. Perhaps we can avoid introducing an the `setM⁻¹!` function? E.g we can initialise `adaptor` with the user-specified mass matrix when it is constructed, instead of setting it in the sampling step? 

```julia
n_adapts, D = 2_000, 10
mass_init = fill(0.5, D)
int = Leapfrog(find_good_eps(h, θ_init))

# initialize mass matrix when `adaptor` is created
metric = DiagEuclideanMetric(mass_init) 
adaptor = StanHMCAdaptor(n_adapts, Preconditioner(metric), NesterovDualAveraging(0.8, int.ϵ))
```

This would make the code slightly cleaner IMO.

Reply:
This is actually a better design! I will try to change the code for this!

Reply:
As a side note, this requires the user to use `Preconditioner(metric)` to create the preconditioner. But this is fine as this is the most common way to create a preconditioner.